src/components/cuda_pw/cupti_component.c

```diff
@@ -0,0 +1,95 @@
+#include <papi.h>
+#include <papi_internal.h>
+#include <papi_vector.h>
+
+#include "debug_comp.h"
+#include "api_cuda.h"
+#include "common_defs.h"
+#include "cuda_common.h"
+#include "cupti_component.h"
+
+#if API_PERFWORKS
+#include "cupti_profiler.h"
+#elif API_EVENTS
+#include "cupti_events.h"
+#endif
+
+extern papi_vector_t _cuda_pw_vector;
+
+int cupti_shutdown(void)
+{
+    return PAPI_OK;
+}
+
+int cupti_init(const char **pdisabled_reason)
+{
+    int retval;
+#if API_PERFWORKS
```

Comment:
Can you use `if defined(API_PERFWORKS)` instead?

Reply:
Not this case per se, but for some of the other functions, this won't be a `#if`, it will be a normal `if` to perform more fine grained runtime branching. But for now, I can make them `#if defined`.

Reply:
I have changed these to #if defined, but actually they are defined as constants `1` or `0`. So they still could be used in fine grained branching.